lib/ansible/modules/cloud/vmware/vmware_vm_host_drs_rule.py

```diff
@@ -0,0 +1,467 @@
+    - Required if C(state) is set to C(present).
+    - These VMs will be added to given VM group name.
+  esxi_hostnames:
+    description:
+    - List of host system names for which DRS rule needs to applied.
+    - Required if C(state) is set ot C(present).
+    - These Host Systems will be added to given host group name.
+  drs_rule_name:
+    description:
+    - The name of the DRS rule to manage.
+    required: True
+  enabled:
+    description:
+    - If set to C(True), the DRS rule will be enabled.
+    - Effective only if C(state) is set to C(present).
+    default: False
+  mandatory:
+    description:
+    - If set to C(True), the DRS rule will be mandatory.
+    - Effective only if C(state) is set to C(present).
+    default: False
+  affinity_rule:
+    description:
+    - If set to C(True), the DRS rule will be an Affinity rule.
+    - If set to C(False), the DRS rule will be an Anti-Affinity rule.
+    - Effective only if C(state) is set to C(present).
+    default: False
+  state:
+    description:
+    - If set to C(present), then the DRS rule is created if not present.
+    - If set to C(present), then the DRS rule is deleted and created if present already.
+    - If set to C(absent), then the DRS rule is deleted if present.
+    required: False
+    default: present
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Create DRS Affinity Rule for VM-Host
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    esxi_hostnames:
+        - DC0_H0
+    drs_rule_name: vm1-vm2-host1-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: True
+
+- name: Create DRS Anti-Affinity Rule for VM-Host
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    esxi_hostnames:
+        - DC0_H0
+    drs_rule_name: vm1-vm2-host1-anti-affinity-rule-002
+    enabled: True
+    mandatory: True
+    affinity_rule: False
+
+- name: Delete DRS Affinity Rule for VM-VM
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    drs_rule_name: vm1-vm2-host1-affinity-rule-001
+    state: absent
+'''
+
+RETURN = r'''#'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec, wait_for_task,
+                                         find_vm_by_id, find_cluster_by_name)
+
+
+class VmwareDrs(PyVmomi):
```

Comment:
Same classname as with the vm-vm rules, not sure if it might conflict/cause issues

Reply:
Done.